src/crypto/secretstream/secretstream_macros.rs

```diff
@@ -128,7 +129,7 @@ pub fn gen_key() -> Key {
 pub struct Stream<M: StreamMode> {
     state: $state_name,
     finalized: bool,
-    phantom: core::marker::PhantomData<M>,
```

Comment:
`core::marker` is not an old Rust it is the way to use sodiumoxide in a no-std env